packages/websockets/context/ws-context-creator.ts

```diff
@@ -109,6 +108,9 @@ export class WsContextCreator {
     };
 
     return this.wsProxy.create(async (...args: unknown[]) => {
+      Object.assign(args[0] ?? {}, {
+        getPattern: () => this.reflectCallbackPattern(callback),
+      });
```

Comment:
```suggestion
      Object.assign(args[0] ?? {}, {
        getPattern: () => this.reflectCallbackPattern(callback),
      });
```
Hmm.. I'm wondering if this won't cause some unexpected issues 🤔 If socket represents a single connection (and so its instance object is shared?), there's a probability that 2 messages from that socket might be processed asynchronously (independently). 

Example:
- Connection to server is established (socket instance is created)
- Message A is emitted 
 - we mutate the "socket instance" enhancing it with the "getPattern()" method
 - before this message is processed, some asynchronous operation is triggered in (guard/interceptor/wherever)
- In the meantime, message B is emitted
 - we mutate the same "socket instance" again replacing the previous "getPattern()" method
- Async operation (mentioned above) completes and we're back to processing message A.
- In another interceptor/guard we call `client.getPattern()` when processing message A but it's already overridden with the message B's "getPattern" implementation

Reply:
Rather than mutating the client, could I add a new `args` entry and let `getPattern()` retrieve `args[2]`? Would that be more "stable"?

Reply:
Great call by the way! I moved `getPattern` to be a method on `WsArgumentHost` instead of under the `getClient()` so that it should be unique per request as the `WsArgumentHost` already is. Tests are still passing and the interfaces are updated. Let me know if you think of any other issues with this approach :smile_cat: